### PR TITLE
tkt-77574: Add assign_localhost prop

### PR DESCRIPTION
**Pull request by skarekrow**
This property will assign a localhost of 127.0.1.1 to all jails that have this property enabled. It will also change the /etc/hosts entry to reflect this change if you set it during creation, otherwise it is manual.

- Also allow True/False for truthy props.

FreeNAS Ticket: #77574

### CODE REVIEW

File: iocage_lib/ioc_common.py [hunk 1]
```diff
@@ -906,7 +906,7 @@ def get_jail_freebsd_version(path, release):
 
 def check_truthy(value):
     """Checks if the given value is 'True'"""
-    if str(value) in ('1', 'on', 'yes', 'true'):
+    if str(value).lower() in ('1', 'on', 'yes', 'true', 'True'):
```
Comment (sonicaj) [inline iocage_lib/ioc_common.py]: If we convert it into `lower`, `True` is not needed ;)

File: iocage_lib/ioc_json.py [hunk 2]
```diff
@@ -1018,6 +1018,10 @@ def check_config(self, conf, default=False):
         if not conf.get('exec_created'):
             conf['exec_created'] = '/usr/bin/true'
 
+        # Version 21 keys
```
Comment (sonicaj) [inline iocage_lib/ioc_json.py:1021]: version bump ;)

File: iocage_lib/ioc_start.py [hunk 3]
```diff
@@ -298,6 +299,14 @@ def __start_jail__(self):
 
                 net.append(f'ip6.addr={ip6_addr}')
 
+            if assign_localhost:
+                # Make sure this exists, jail(8) will tear it down if we don't
+                # manually do this.
+                if self.check_aliases('127.0.1.1', '4') != '127.0.1.1':
+                    su.run(['ifconfig', 'lo0', 'alias', '127.0.1.1/32'])
+
+                net.append(f'ip4.addr=127.0.1.1')
```
Comment (sonicaj) [inline iocage_lib/ioc_start.py]: We would be writing `ip4.addr` twice i think, is that desirable ?

Comment (skarekrow) [reply iocage_lib/ioc_start.py]: Yes that's fine, jail will only apply it once

**Review by sonicaj — APPROVED** [review]

**Comment by skarekrow** [comment]
Unrelated failure

**Pull request merged**